src/v/cluster/data_migration_backend.cc

```diff
@@ -504,28 +531,43 @@ ss::future<errc> backend::do_topic_work(
 ss::future<errc> backend::do_topic_work(
   const model::topic_namespace& nt,
   state sought_state,
-  const outbound_topic_work_info&) {
-    // this assert is in accordance to the logic in get_work_scope
-    vassert(
-      sought_state == state::finished,
-      "only ->finished state transition requires topic work");
-    ss::abort_source as;
-    retry_chain_node rcn{as, ss::lowres_clock::time_point::max(), 2s};
-    // todo: unmount first
-    auto unmount_res = co_await retry_loop(
-      rcn, [this, &nt, &rcn] { return unmount_topic(nt, rcn); });
-    if (unmount_res == errc::topic_not_exists) {
-        vlog(dm_log.warn, "topic {} does not exist, ignoring", nt, unmount_res);
+  const outbound_topic_work_info&,
+  tsws_lwptr_t tsws) {
+    auto& rcn = tsws->rcn;
+    // this switch should be in accordance to the logic in get_work_scope
+    switch (sought_state) {
+    case state::finished: {
+        // unmount first
+        auto unmount_res = co_await unmount_topic(nt, rcn);
+        if (unmount_res != errc::success) {
+            vlog(
+              dm_log.warn, "failed to unmount topic {}: {}", nt, unmount_res);
+            co_return unmount_res;
+        }
+        // delete
+        co_return co_await delete_topic(nt, rcn);
+    }
+    case state::cancelled: {
```

Comment:
Just to make sure i understand this. The topic partitions are unmount only after the migration state is `cut_over`, which can not be cancelled ?

Reply:
Yes, that's right. I'll add it to the comment.